macros/edr/data_monitoring/anomaly_detection/get_anomaly_scores_query.sql

```diff
@@ -83,6 +83,11 @@
                 source_value,
                 bucket_start,
                 bucket_end,
+                {%- if metric_properties and metric_properties.seasonality and metric_properties.seasonality=='day_of_week' %}
```

Comment:
Repeating my comment on this:

I think it's a pretty narrow name and implementation.
If we want the flexibility to support other seasonality configs in the future, I suggest:

1. Naming the column `bucket_seasonality`
I think it would also be much clearer to save there Sunday instead of 1, and null instead of 0.

2. Set a var in the top like:
`{%- set seasonality_expression = get_bucket_seasonality() %}`
And do the logic of returning 0/null or an expression there (easier to expand later and also looks clearer, I think).

Reply:
Also now that you are not using modulo, for sure this is a bad column name

Reply:
> bucket_seasonality 
This name is so much better! 

> much clearer to save there Sunday instead of 1, and null instead of 0.
The 0 in case of no seasonality is just so we don't have to deal with differences in the DWH behaviors when partitioning by a None.
 
Do you think it can be a constant string without losing that better readability? "no_seasonality" or some form of that? 

About the days of the week vs the names of the days - I think day of week as number 0-6 is a more natural implementation than the long form of the name. This is a function that exists in all DWHs and versions, whereas getting the long name of the day of the week is something that's a bit less supported. For example in Spark dayofweek function exists, but to get the longform name this is the function call needed : `date_format(my_timestamp, 'EEEE')` (from this SO [thread](https://stackoverflow.com/a/33764798/20981468) )

Do you think this column would be more readable if it were to have the seasonality expression inside it, something like 
```
'{"seasonality":"day_of_week", "day_of_week":0}'
'{"seasonality":"day_of_week", "day_of_week":1}'
'{"seasonality":"day_of_week", "day_of_week":2}'
'{"seasonality":"day_of_week", "day_of_week":3}'
```
? 
And then the no seasonality case could look like
```
'{"seasonality":"none"}'
```

WDYT?

Reply:
using the name of the day for the bucket for cases where we have seasonality, and the string `no_seasonality` for cases where we don't have seasonality.